indico/modules/rb_new/client/js/common/bookings/BookingEditForm.jsx

```diff
@@ -43,10 +44,14 @@ class BookingEditForm extends React.Component {
         booking: PropTypes.object.isRequired,
         formProps: PropTypes.object.isRequired,
         onBookingPeriodChange: PropTypes.func,
+        hideOptions: PropTypes.objectOf(PropTypes.bool),
+        optionalFields: PropTypes.objectOf(PropTypes.bool),
```

Comment:
I'd use a simple `reasonRequired` bool (default true) here. I think `reason` is the only field where making it optional makes any sense or even works (at least if we ever make `booked_for_id` NOT NULL as it should be)